src/Umbraco.Infrastructure/Migrations/Upgrade/V_11_4_0/AlterKeyValueDataType.cs

```diff
@@ -0,0 +1,29 @@
+using System.Reflection;
+using NPoco;
+using Umbraco.Cms.Infrastructure.Persistence.Dtos;
+
+namespace Umbraco.Cms.Infrastructure.Migrations.Upgrade.V_11_4_0;
+
+public class AlterKeyValueDataType : MigrationBase
+{
+    public AlterKeyValueDataType(IMigrationContext context)
+        : base(context)
+    { }
+
+    protected override void Migrate()
+    {
+        string tableName = KeyValueDto.TableName;
+        string colName = "value";
+
+        // Determine the current datatype of the column within the database
+        string colDataType = Database.ExecuteScalar<string>($"SELECT TOP(1) CHARACTER_MAXIMUM_LENGTH FROM INFORMATION_SCHEMA.COLUMNS" +
+            $" WHERE TABLE_NAME = '{tableName}' AND COLUMN_NAME = '{colName}'");
```

Comment:
Hi there @Rockerby! I was suspecting this already but unfortunately this code won't work on SQLite:

```
Umbraco.Cms.Core.Install.InstallException: The database failed to upgrade. ERROR: The database configuration failed with the following message: SQLite Error 1: 'no such table: INFORMATION_SCHEMA.COLUMNS'.
 Please check log file for additional information (can be found in 'LoggingSettings.Directory')
   at Umbraco.Cms.Infrastructure.Install.InstallSteps.DatabaseUpgradeStep.ExecuteAsync(Object model) in D:\Dev\Umbraco-CMS11\src\Umbraco.Infrastructure\Install\InstallSteps\DatabaseUpgradeStep.cs:line 57
   at InvokeStub_InstallSetupStep`1.ExecuteAsync(Object, Object, IntPtr*)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```

Since the type of the column in SQLite is already `TEXT` (no 255 limit), I think it's safe to say in this migration that if it's SQLite then immediately return from the `Migrate()` method, no work is needed.

Reply:
Thanks for the testing on this one @nul800sebastiaan, I've added in a check to see if we're using SQL Lite, and not to run the change if we are 👍

Ahead of your tests, I've ran both a fresh install and an upgrade path to this (on both SQL Lite and SQL Server) and everything is working as expected.